standards/core/rln-contract.md

```diff
@@ -45,27 +45,37 @@ The contract MUST provide the following functionalities:
-A membership _keeper_ is the entity that controls the Ethereum address used to register that membership.
+A membership _keeper_ is the sender of the transaction that registered that membership.
+Transaction sender in this context is defined as `msg.sender` in Solidity semantics.
+The contract MUST support transactions sent directly from externally-owned accounts (EOA).
+The contract MAY support transactions sent from an EOA via a chain of contract calls,
+in which case the last contract in the call chain MAY be designated as the membership keeper.
+The contract MAY also support meta-transactions sent via paymasters or relayers,
+which MAY require additional authentication-related logic.
+
 The holder and the keeper MAY be different entities for the same membership.
 When authorizing membership-related requests,
 the contract SHOULD distinguish between the keeper and non-keepers,
 and MAY also use additional criteria.
 
 Contract parameters and their RECOMMENDED values for the initial mainnet deployment are as follows:
 
-| Parameter                                               | Symbol    | Value    | Units                |
-| ------------------------------------------------------- | --------- | -------- | -------------------- |
-| Epoch length                                            | `epoch`   | `10`     | minutes              |
-| Maximum total rate limit of all memberships in the tree | `R_{max}` | `160000` | messages per `epoch` |
-| Minimum rate limit of one membership                    | `r_{min}` | `20`     | messages per `epoch` |
-| Maximum rate limit of one membership                    | `r_{max}` | `600`    | messages per `epoch` |
-| Membership price for `1` message per epoch              | `p_u`     | `0.05`   | `USD`                |
-| Membership expiration term                              | `T`       | `180`    | days                 |
-| Membership grace period                                 | `G`       | `30`     | days                 |
-| Accepted tokens                                         |           | `DAI`    |                      |
+| Parameter                                               | Symbol    | Value    | Units              |
+| ------------------------------------------------------- | --------- | -------- | ------------------ |
+| Epoch length                                            | `t_{ep}`  | `600`    | seconds            |
+| Maximum total rate limit of all memberships in the tree | `R_{max}` | `160000` | messages per epoch |
+| Minimum rate limit of one membership                    | `r_{min}` | `20`     | messages per epoch |
+| Maximum rate limit of one membership                    | `r_{max}` | `600`    | messages per epoch |
+| Membership price for `1` message per epoch              | `p_u`     | `0.05`   | `USD`              |
+| Membership expiration term                              | `T`       | `180`    | days               |
```

Comment:
I would suggest renaming `Membership expiration term` to `Membership billing period` or something similar + slightly changing its definition.

As it is, it gives the impression that your membership will expire after `Membership expiration term` aka `T` aka `180days`. However, I think we should allow the user to register for several *billing periods*.

This can actually allow lowering the `T` from `180days` to eg `30days` which is a more common period (rent, salary, netflix) + it will be less pricy for a user wanting to test the protocol, without having to commit for the price of 6 months. But if the user wishes, he can also register for an arbitrary number of periods.

I would also change the:
* `Membership price for 1 message per epoch` aka `p_u`
* to `Membership price for 1 message per epoch per period` aka `p_u`

And as an implementation detail cc @richard-ramos allow membership registration for an arbitrary number of periods. Where price is calculated as `num_periods * p_u * num_messages`.


In other words, I think we should:
* Allow registering for n amount of periods.
* Perhaps lower 180, since `180*n` doesn't give much granularity

Reply:
For grace period, we can either multiply it for the number of periods, (or convert it to a %).
For example, assuming the original value of expiration term of 180 days and 30d grace period, if we change it to periods of 30d, the equivalent grace period would be ~5d, so anyone registering for 6 periods would end up with 30 of grace period (5d*6).

Reply:
Should we have a maximum number of periods an user could register as a way of preventing squatting of the slots on the merkle tree or is this a non-issue?

Reply:
Another question is,  how would extending a membership work in this case? should they also specify the number of periods they want to extend?

Reply:
> Should we have a maximum number of periods an user could register as a way of preventing squatting of the slots on the merkle tree or is this a non-issue?

yes, i think so. eg max 1 or 2 years. this limit is indeed necessary because if someone registers for 10000 years, thats a liability for waku, since waku would need to provide services for 10000 years?

> Another question is, how would extending a membership work in this case? should they also specify the number of periods they want to extend?

yes.

Reply:
I would keep a max to 6 months as we are still early stage.

Reply:
As discussed on research call 2024-09-09, I've created an [issue](https://github.com/waku-org/specs/issues/37) in the spec repo and am resolving this discussion for now. Let's consider this idea more thoroughly for the next iteration of the spec.

> renaming Membership expiration term to Membership billing period or something similar

I'm also adding this terminology clarification to this PR.